agent/agent.c

```diff
@@ -1557,8 +1557,11 @@ nice_agent_gather_candidates (NiceAgent * agent, guint stream_id)
           GST_WARNING_OBJECT (agent,
               "%u/%u: Unable to gather host candidate for address %s",
               stream->id, component->id, ip);
-          ret = FALSE;
-          goto error;
+          if (configured_local_addresses){
+            /* The client specified these addresses, so treat this as an error! */
+            ret = FALSE;
+            goto error;
+          }
         }
 
```

Comment:
Needs an else clause here as if we fall through with udp_host_candidate == NULL the following code to get stun and turn candidates will segault trying to access sockptr

Reply:
I feared that would happen initially, but I'm seetting borh turn and stun while testing, and it didn't crash. I'll take a second look!